File: src/main/java/dkoryakin/aston/demo/domain/Account.java
package dkoryakin.aston.demo.domain;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;

@Data
@Builder
public class Account {

    private Long id;

    private String name;

    private Pin pin;

    private BigDecimal balance;

    public void deposit(BigDecimal amount) {
        balance = balance.add(amount);
    }

    public void withdraw(BigDecimal amount) {
        balance = balance.subtract(amount.abs());
    }

    public void transfer(BigDecimal amount, Account destination) {
        balance = balance.subtract(amount.abs());
        destination.balance = balance.add(amount);
    }
}

File: src/main/java/dkoryakin/aston/demo/domain/Account.kt
package dkoryakin.aston.demo.domain

import kotlin.math.abs

data class Account(private val id: Long,
        private val name: String,
        private val pin: Pin, var balance: Double) {

    fun deposit(amount: Double) {
        balance += amount
    }

    fun withdraw(amount: Double) {
        balance -= abs(amount)
    }

    fun transfer(amount: Double, destination: Account) {
        balance -= amount
        destination.balance += amount
    }
}
